test/tap/shrinkwrap-lifecycle.js

```diff
@@ -0,0 +1,42 @@
+var fs = require('graceful-fs')
+var path = require('path')
+
+var mkdirp = require('mkdirp')
+var osenv = require('osenv')
+var rimraf = require('rimraf')
+var test = require('tap').test
+
+var common = require('../common-tap.js')
+var pkg = path.resolve(__dirname, 'shrinkwrap-lifecycle')
+
+test('npm shrinkwrap execution order', function (t) {
+  setup()
+  fs.writeFileSync(path.resolve(pkg, 'package.json'), JSON.stringify({
+    author: 'Simen Bekkhus',
+    name: 'shrinkwrap-lifecycle',
+    shrinkwrap: '0.0.0',
+    description: 'Test for npm shrinkwrap execution order',
+    scripts: {
+      preshrinkwrap: 'echo this happens first',
+      postshrinkwrap: 'echo this happens second'
```

Comment:
But it needs to do something, as every other lifecycle includes all three. What it would do, I dunno, usually in cases like this its called after, just before the `post` lifecycle.